crates/zksync/core/src/vm/storage_view.rs

```diff
@@ -27,6 +33,11 @@ pub(crate) struct StorageView<S> {
     initial_writes_cache: HashMap<StorageKey, bool>,
     /// The tx caller.
     caller: H160,
+    /// Call tracker for recording storage accesses.
+    /// Track `FarCalls`s to allow matching them with their respective `Ret` opcodes.
+    /// zkEVM erases the `msg.sender` and `code_address` for certain calls like to MsgSimulator,
```

Comment:
Wonder if this is still possible to retrieve... This is MimicCall's doing, right?

Reply:
The `Ret` opcode has already popped the stack, so this information is lost or must be recovered from `before_execution`, but its no longer present since the registers (holding msg.sender) are cleared after the `FarCall` opcode execution to make way for new opcodes.